packages/okta-react/src/AuthService.js

```diff
@@ -196,12 +190,24 @@ class AuthService {
   }
 
   async login(fromUri, additionalParams) {
+    if(this._pending.handleLogin) { 
+      // Don't trigger second round
+      return;
+    }
+
+    this._pending.handleLogin = true;
+    // Update UI pending state
+    this.emitAuthState({ ...this.getAuthState(), isPending: true });
     // Save the current url before redirect
     this.setFromUri(fromUri); // will save current location if fromUri is undefined
-    if (this._config.onAuthRequired) {
-      return this._config.onAuthRequired(this);
+    try {
+      if (this._config.onAuthRequired) {
+        return await this._config.onAuthRequired(this);
+      }
+      return await this.redirect(additionalParams);
```

Comment:
Errors are just silently lost - is there anything there we'd want to inform the user of?

Reply:
only finally been added here, but without catch, I think users still can catch errors if any happen.

Reply:
*scrunches face up in confusion*
*writes test code*
*blinks furiously*
*writes more test code*
*googles*
Huh.  This is not how I expected finally and errors to interact.  I sit corrected.

Reply:
Test added.